SURF-449 Add documentation for GQL FOR as alternate syntax for UNWIND (#1503)
neo4j/docs-cypher: modules/ROOT/pages/clauses/unwind.adoc

The `UNWIND` clause makes it possible to transform any list back into individual rows.
These lists can be parameters that were passed in, previously `collect`-ed result, or other list expressions.

[NOTE]
====

> As of Neo4j 2026.03, the same semantics are also available with GQL-aligned syntax: xref:clauses/for.adoc[`FOR variable IN expression`].

Reviewer: ```suggestion
As of Neo4j 2026.03, the same semantics is also available with GQL conformant syntax: xref:clauses/for.adoc[`FOR variable IN expression`].
```

"GQL conformant" is the term the docs use. "GQL-aligned" does not appear anywhere yet. "Conformance" is also what the standard itself calls it.

I leave it to @rsill-neo4j to decide if it should be "GQL-conformant syntax" or "GQL conformant syntax". I only point out that "GQL conformant" currently appears in the docs while "GQL-conformant" does not.

And:

<img width="745" height="567" alt="Image" src="https://github.com/user-attachments/assets/8cdba4bb-4b3f-45e1-b67f-4933cc7b1212" />

> As of Neo4j 2026.03, the same semantics is also available with GQL conformant syntax: xref:clauses/for.adoc[`FOR variable IN expression`].

Reviewer 2: 2026.04

> As of **Neo4j 2026.03**, the same semantics are also available with GQL-aligned syntax: xref:clauses/for.adoc[`FOR variable IN expression`].

Reviewer 3: ```suggestion
As of Neo4j 2026.03, the same semantics are also available with GQL-aligned syntax: xref:clauses/for.adoc[`FOR variable IN expression`].
```
